websocket_test.go

```diff
@@ -667,6 +667,70 @@ func TestWebSocket_Streaming_Parallel(t *testing.T) {
 	}
 }
 
+// Tests the correct returning of values depending on the ParallelOptions structure
+func TestParallelOptions_GetList(t *testing.T) {
+	l := NewLocalTest(tSuite)
+	defer l.CloseAll()
+
+	var po *ParallelOptions
+	_, roster, _ := l.GenTree(3, false)
+	nodes := roster.List
+
+	count, list := po.GetList(nodes)
+	require.Equal(t, 2, count)
+	require.Equal(t, 3, len(list))
+	require.False(t, po.Quit())
+
+	po = &ParallelOptions{}
+	count, list = po.GetList(nodes)
+	require.Equal(t, 2, count)
+	require.Equal(t, 3, len(list))
+	require.False(t, po.Quit())
+
+	first := 0
+	for i := 0; i < 32; i++ {
+		_, list := po.GetList(nodes)
+		if (<-list).Equal(nodes[0]) {
+			first++
+		}
+	}
+	require.NotEqual(t, 0, first)
+	require.NotEqual(t, 32, first)
+	po.DontShuffle = true
+	first = 0
+	for i := 0; i < 32; i++ {
+		_, list := po.GetList(nodes)
+		if (<-list).Equal(nodes[0]) {
+			first++
+		}
+	}
+	require.Equal(t, 32, first)
+
+	po.IgnoreNodes = append(po.IgnoreNodes, nodes[0])
+	count, list = po.GetList(nodes)
+	require.Equal(t, 2, count)
+	require.Equal(t, 2, len(list))
+
+	po.IgnoreNodes = append(po.IgnoreNodes, nodes[1])
+	count, list = po.GetList(nodes)
+	require.Equal(t, 2, count)
+	require.Equal(t, 1, len(list))
+
+	po.IgnoreNodes = po.IgnoreNodes[0:1]
+	po.QuitError = true
+	require.True(t, po.Quit())
+
+	po.AskNodes = 1
+	count, list = po.GetList(nodes)
+	require.Equal(t, 1, count)
+	require.Equal(t, 1, len(list))
+
+	po.StartNode = 1
+	count, list = po.GetList(nodes)
+	require.Equal(t, 1, count)
+	require.Equal(t, 1, len(list))
+}
+
```

Comment:
could you add a test that runs `SendProtobufParallel`? just to make sure it continues to work in the future